src/full/Agda/TypeChecking/Monad/Signature.hs

```diff
@@ -13,12 +13,15 @@ import Control.Monad.Writer
 import Control.Monad.Trans.Maybe
 import Control.Monad.Trans.Identity
 
+import Data.Foldable (for_)
 import qualified Data.List as List
+import qualified Data.List.NonEmpty as NonEmpty
 import Data.Set (Set)
 import qualified Data.Set as Set
 import qualified Data.Map as Map
 import qualified Data.HashMap.Strict as HMap
 import Data.Maybe
+import qualified Data.Semigroup as S
```

Comment:
I'd prefer an unqualified import of `Data.Semigroup`.  At some point, these imports will then simply be redundant as Haskell is heading towards having it in the prelude.